src/upnp_cds.cc

```diff
@@ -93,15 +93,15 @@ void ContentDirectoryService::doBrowse(const std::unique_ptr<ActionRequest>& req
     if (config->getBoolOption(CFG_SERVER_HIDE_PC_DIRECTORY))
         flag |= BROWSE_HIDE_FS_ROOT;
 
-    auto param = std::make_unique<BrowseParam>(objectID, flag);
+    auto param = BrowseParam(objectID, flag);
 
-    param->setStartingIndex(stoiString(startingIndex));
-    param->setRequestedCount(stoiString(requestedCount));
-    param->setSortCriteria(trimString(sortCriteria));
+    param.setStartingIndex(stoiString(startingIndex));
+    param.setRequestedCount(stoiString(requestedCount));
+    param.setSortCriteria(trimString(sortCriteria));
 
     std::vector<std::shared_ptr<CdsObject>> arr;
     try {
-        arr = database->browse(param);
+        arr = database->browse(std::make_unique<BrowseParam>(param));
```

Comment:
I'm quite sure this breaks browsing, because browse calls `param->setTotalMatches` which is read further down and I assume this call creates a copy of `param`.

Reply:
yes it does.

Reply:
Removed.